idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/folding/foldingFileEditorListener.kt

```diff
@@ -0,0 +1,53 @@
+package arrow.meta.ide.plugins.proofs.folding
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.phases.ExtensionPhase
+import com.intellij.codeInsight.folding.CodeFoldingManager
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.EditorFactory
+import com.intellij.openapi.editor.event.CaretEvent
+import com.intellij.openapi.editor.event.CaretListener
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.fileEditor.FileEditor
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileEditor.FileEditorProvider
+import com.intellij.openapi.util.Pair
+import com.intellij.openapi.vfs.VirtualFile
+
+val IdeMetaPlugin.foldingFileEditor: ExtensionPhase
+  get() = addFileEditorListener(
+    selectionChanged = {},
+    fileOpened = { manager: FileEditorManager, file: VirtualFile ->
+      val fileEditor = manager.getSelectedEditor(file)
+      val document = FileDocumentManager.getInstance().getDocument(file)
```

Comment:
Lift those to the parameter

Reply:
aren't those specific to our use case ?? Maybe those are not needed for other use cases, right?